pkg/github/github.go

```diff
@@ -141,16 +184,27 @@ func componentFromRepo(
 		return component, err
 	}
 
-	if repo.Version == "" {
-		highestVersion, err := version.HighestVersion(availableVersions)
-		if err != nil {
-			return component, err
-		}
+	highestVersion, err := version.HighestVersion(availableVersions)
+	if err != nil {
+		return component, err
+	}
 
+	if repo.Version == "" {
 		log.OutLogger.Printf("  Repo 'latest' release resolved as '%s'", highestVersion)
 		repo.Version = highestVersion
 	}
 
+	// Get a comparison between the highest version and HEAD
+	comparison, err := compareRefs(httpClient, repo.Name, highestVersion, "HEAD")
```

Comment:
I think we should also included test cases for these changes to componentFromRepo

Reply:
@BradleyBoutcher I agree with you. The challenge here is that componentFromRepo is being tested in a somewhat implicit way in `cli_test.go`. These tests are making actual calls to the Github API. You'll note that we're testing the following formats `"changelog", "docs-release", "release"` but not `unreleased`, and that's because `unreleased` is the only format that results in dynamic output from the API. 

I think that we should really be testing componentFromRepo in isolation and we should be careful to structure things in such a way that the http-call making code is isolated or potentially dependency injected so that the logic of the method can be tested without relying on calls to the actual Github API. I think that bit of work is out of scope for this story though.

Reply:
I totally agree. I think one of the main problems with this code base is that we have more indirect testing than direct testing, and maybe we should have a larger conversation about that. As it gets bigger, I'm worried that it'll become more difficult to fix.

Reply:
RunParser would, in my opinion, be the best place to start with dependency injection so we can start testing each piece individually. From there, we can start making each test for the dependencies more fine grained. If it's within the scope of our current epic, I'd be happy to take that on.

Reply:
we can log a follow-up issue, but it's out of the scope of the current epic for now

we're so close to wrapping up the work in the epic, though, that if all but XA / releasing is done, we could pick that card up.